### ISSUE TITLE
I want to ask some questions

### ISSUE THREAD

**Issue by xdmdtz123** [ralphabb/GNN-RNI#6]
Deer, author:
  Your research is awesome, I've read some of your codes and wanted to ask you some questions.
  First, I found two suspicious little bugs.

1. When the code runs in the GPU environment, an error will be reported, so I think the following modifications should be made.

![图片](https://user-images.githubusercontent.com/61136601/157286693-fa9565e0-d8a9-41ef-be84-404fd9db2742.png)

2.I think num_ Features should be changed to num_ node_ features. Although this does not affect the normal operation of the program.
![图片](https://user-images.githubusercontent.com/61136601/157289322-ffa65dba-818a-464d-b094-5b0561dec09b.png)

 Second, this is the first time I read a paper related to graph neural network. I don't feel that I have a deep understanding, so I drew a diagram to ask you whether I understand it or not.
![图片](https://user-images.githubusercontent.com/61136601/157289871-9afe09ff-3e28-4de0-9080-242a0c3f9e8c.png)

Third, when I run "python GNNHyb.py -dataset CEXP -layers 3 -randomRatio 0.85 -probDist u", the result is poor.
![图片](https://user-images.githubusercontent.com/61136601/157290596-b6c81b0d-cd4f-432e-88f2-e22b53cb6362.png)
I ran the program according to your method. As the number of iterations increases, no matter how much randomratio is set, the running result is almost unchanged.I don't know what's wrong.

**Comment by ralphabb** [comment]
Hi, 
Thanks for getting in touch!
1. Makes sense, but I don't quite understand 2. The point is to map from the 2-dimensional one-hot encodings, so I'm not sure what this change is doing. 
As for your understanding, it seems correct: We concatenate RNI at the very beginning to the node features, and then perform message passing over them. Finally, as for your results, this makes sense, as 3 layers are insufficient to properly propagate information along the graph structure we have. The command is simply intended as an example.  In the original paper, we use 8 message passing iterations (Page 4: https://arxiv.org/pdf/2010.01179.pdf). Hence, you probably should try the hyper-parameters reported in the paper (8 layers, normal distribution for RNI).

**Comment by xdmdtz123** [comment]
I rerun the code referring to the hyper-parameters table in the appendix to the article. The accuracy is amazing, as described in the article. Thank you for your reply.
